pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,206 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	"kubevirt.io/client-go/kubecli"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	ipamclaims "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+)
+
+type NetConf struct {
+	AllowPersistentIPs bool `json:"allowPersistentIPs,omitempty"`
+}
+
+func GetNetworkToIPAMClaimName(virtClient kubecli.KubevirtClient, namespace string, name string, networks []virtv1.Network, persistentIPsEnabled bool) (map[string]string, error) {
+	networkToIPAMClaimName := make(map[string]string)
+	for _, network := range networks {
+		if vmispec.IsSecondaryMultusNetwork(network) {
+			allowPersistentIPs, err := getAllowPersistentIPs(virtClient, namespace, network.Multus.NetworkName)
+			if err != nil {
+				return map[string]string{}, fmt.Errorf("GetNetworkToIPAMClaimName failed: %w", err)
+			}
+
+			if allowPersistentIPs {
+				networkToIPAMClaimName[network.Name] = fmt.Sprintf("%s.%s", name, network.Name)
+			}
+		}
+	}
+
+	if !persistentIPsEnabled && len(networkToIPAMClaimName) != 0 {
+		return nil, fmt.Errorf("persistenIPs is requested but PersistentIPsGate is disabled")
+	}
+
+	return networkToIPAMClaimName, nil
+}
+
+func getAllowPersistentIPs(virtClient kubecli.KubevirtClient, namespace string, fullNetworkName string) (bool, error) {
+	namespace, networkName := getNamespaceAndNetworkName(namespace, fullNetworkName)
+	nad, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Get(context.Background(), networkName, v1.GetOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to locate network attachment definition %s/%s", namespace, networkName)
+	}
+
+	allowPersistentIPs, err := getAllowPersistentIPsFromNAD(nad)
+	if err != nil {
+		return false, fmt.Errorf("getAllowPersistentIPs failed: %w", err)
+	}
+
+	return allowPersistentIPs, nil
+}
+
+func getAllowPersistentIPsFromNAD(network *networkv1.NetworkAttachmentDefinition) (bool, error) {
+	if network.Spec.Config == "" {
+		return false, nil
+	}
+
+	var netConf NetConf
+	err := json.Unmarshal([]byte(network.Spec.Config), &netConf)
+	if err != nil {
+		return false, fmt.Errorf("failed to unmarshal NAD spec.config JSON: %v", err)
+	}
+
+	return netConf.AllowPersistentIPs, nil
+}
+
+func CreateIPAMClaims(client kubecli.KubevirtClient, vmi *virtv1.VirtualMachineInstance, ownerRef *v1.OwnerReference, persistentIPsEnabled bool) error {
```

Comment:
If persistentIPsEnabled is false we should not even call this, so it do not make much sense to pass the flag.